Add a pass to convert rank-0 tensor to rank-1 tensor

cccclai opened the pull request:
Differential Revision: D69281867


Test with following code, and no long see the error/warning to complain rank-0 tensor
```

class Model(torch.nn.Module):
    def forward(self, x, y):
        return x + y

model = Model()
model.eval()

example_inputs = (torch.tensor(1.0), torch.tensor(2.0))
exported_program_manager_aten = torch.export.export(model, example_inputs)
exported_program_manager_edge = executorch.exir.to_edge(
    exported_program_manager_aten
).transform([Rank0ToRank1Pass()])

delegated_module = to_backend(
    CoreMLBackend.__name__, exported_program_manager_edge.exported_program(), []
)
```

shoumikhin commented:
@pytorchbot label "topic: not user facing"

shoumikhin commented:
@cccclai @YifanShenSZ  I wonder if Core ML is not gonna support scalar tensors, does it make sense to embed such workaround somewhere deeper to happen automatically under the hood when using the Core ML backend?

metascroy commented:
> @cccclai @YifanShenSZ I wonder if Core ML is not gonna support scalar tensors, does it make sense to embed such workaround somewhere deeper to happen automatically under the hood when using the Core ML backend?

Specifically, adding new cases here to cast scalars to tensors: https://github.com/pytorch/executorch/blob/main/backends/apple/coreml/runtime/delegate/coreml_backend_delegate.mm#L72-L95

It looks like ct.convert already does the cast, we just don't have it in CoreML's ET runtime.

cccclai committed:
Add a pass to convert rank-0 tensor to rank-1 tensor (#8298)

Summary:


Test with following code, and no long see the error/warning to complain rank-0 tensor
```

class Model(torch.nn.Module):
    def forward(self, x, y):
        return x + y

model = Model()
model.eval()

example_inputs = (torch.tensor(1.0), torch.tensor(2.0))
exported_program_manager_aten = torch.export.export(model, example_inputs)
exported_program_manager_edge = executorch.exir.to_edge(
    exported_program_manager_aten
).transform([Rank0ToRank1Pass()])

delegated_module = to_backend(
    CoreMLBackend.__name__, exported_program_manager_edge.exported_program(), []
)
```

Differential Revision: D69281867
